### ISSUE TITLE
UncaugthError while enabling dragging on a marker not shown

### ISSUE THREAD

**Issue by teriblus** [Leaflet/Leaflet#5562]
- Leaflet version I'm using: 1.0.3
- Browser (with version) I'm using: chrome 58
- OS/Platform (with version) I'm using: debian jessie

When calling marker.dragging.enable() on a marker that is not shown on the map (never added through addLayer or removed by removeLayer), the following error occurs: 
leaflet.js:5 Uncaught TypeError: Cannot read property 'classList' of null
    at Object.addClass (leaflet.js:5)
    at e.addHooks (leaflet.js:9)
    at enable (leaflet.js:8)
    at HTMLInputElement.document.getElementById.onclick ((index):109)

However, when I'm adding the marker after that, dragging is enabled.

I created a jsfiddle showing the error: https://jsfiddle.net/05d1wqq1/
Steps:
1. open dev tools on chrome
2. click on hide_marker
3. click on enable_dragging
4. error appears in the console

**Comment by rshadow** [comment]
Same error on
```
/* @preserve
 * Leaflet 1.2.0+Detached: 1ac320ba232cb85b73ac81f3d82780c9d07f0d4e.1ac320b, a JS library for interactive maps. http://leafletjs.com
 * (c) 2010-2017 Vladimir Agafonkin, (c) 2010-2011 CloudMade
 */
```

Stack:
```
TypeError: Cannot read property 'classList' of undefined
at pt (https://so.nowtaxi.ru/js/leaflet/leaflet.js:5:8893)
at e._animateZoom (https://so.nowtaxi.ru/js/leaflet/leaflet.js:5:47553)
at e.<anonymous> (https://so.nowtaxi.ru/js/leaflet/leaflet.js:5:47416)
```

And i try find it in sourse:
```
if (el.classList !== undefined) {                               line 2578
addClass(this._mapPane, 'leaflet-zoom-anim');    line 4510
_animateZoom(center, zoom, true);                      line 4496
```

**Comment by perliedman** [comment]
I can confirm that I can get the fiddle linked in the original post to produce two different results:

```
VM252 leaflet.js:5 Uncaught TypeError: Cannot read property 'classList' of null
    at Object.addClass (VM87 leaflet.js:5)
    at e.addHooks (VM87 leaflet.js:9)
    at enable (VM87 leaflet.js:8)
    at HTMLInputElement.document.getElementById.onclick ((index):109)
```

and

```
leaflet.js:6 Uncaught TypeError: Cannot read property '_leaflet_events' of null
    at Object._on (VM87 leaflet.js:6)
    at Object.on (VM87 leaflet.js:6)
    at enable (VM87 leaflet.js:8)
    at e.addHooks (VM87 leaflet.js:9)
    at enable (VM87 leaflet.js:8)
    at HTMLInputElement.document.getElementById.onclick ((index):109)
```

**Comment by perliedman** [comment]
To fix this, we just need to find the approriate level to put `null`/`undefined` guards.

Without having looked closely, it's likely the guards should be in `addHooks`.

**Comment by cherniavskii** [comment]
Duplicate of #5293, fixed by #5295.
Described bug isn't reproducible in Leaflet 1.2.0 version.
Here is forked jsfiddle from first post, but with Leaflet 1.2.0 version: https://jsfiddle.net/x20bt41f/1/
